init.el

```diff
@@ -200,7 +200,19 @@ Also contains along with versions and other config.")
          ;; M-x interface with Ido-style fuzzy matching.
          (:name smex
                 :after (progn (smex-initialize)
-                              (global-set-key (kbd "M-x") 'smex))))
+                              (global-set-key (kbd "M-x") 'smex)))
+
+         ;; Major mode for JSX and JS files
+         (:name rjsx-mode
+                :after (progn (add-to-list 'auto-mode-alist '("\\.js\\'" . rjsx-mode))
+                              (add-to-list 'auto-mode-alist '("\\.jsx\\'" . rjsx-mode))
+                              (setq js2-basic-offset 2
+                                    js-switch-indent-offset 2)))
+
+         ;; Format JS, JSX files on save event.
+         ;; Prerequisite: npm install -g prettier`
+         (:name prettier-js
+                :after (add-hook 'rjsx-mode-hook 'Prettier-Js-mode)))
 
```

Comment:
Accepting this merge commit for the moment, but you can modify this code in two ways: 

1. For prettier-js, add the hook to rjsx mode after rjsx mode is installed (using `eval-after-load`)
2. Either throw an error if prettier is not installed, or add the hook only if prettier is installed, depending on the behaviour you want. (use `executable-find` for this)